go/vt/vttablet/queryservice/queryservice.go

```diff
@@ -109,78 +107,32 @@ type QueryService interface {
 
-	ReserveStreamExecute(ctx context.Context, target *querypb.Target, preQueries []string, sql string, bindVariables map[string]*querypb.BindVariable, transactionID int64, options *querypb.ExecuteOptions, callback func(*sqltypes.Result) error) (int64, *topodatapb.TabletAlias, error)
+	ReserveStreamExecute(ctx context.Context, target *querypb.Target, preQueries []string, sql string, bindVariables map[string]*querypb.BindVariable, transactionID int64, options *querypb.ExecuteOptions, callback func(*sqltypes.Result) error) (ReservedState, error)
 
 	Release(ctx context.Context, target *querypb.Target, transactionID, reservedID int64) error
 
 	// Close must be called for releasing resources.
 	Close(ctx context.Context) error
 }
 
-type resultStreamer struct {
-	done chan struct{}
-	ch   chan *sqltypes.Result
-	err  error
-}
-
-func (rs *resultStreamer) Recv() (*sqltypes.Result, error) {
-	select {
-	case <-rs.done:
-		return nil, rs.err
-	case qr := <-rs.ch:
-		return qr, nil
-	}
+type TransactionState struct {
+	TransactionID int64
+	TabletAlias   *topodatapb.TabletAlias
 }
 
-// ExecuteWithStreamer performs a StreamExecute, but returns a *sqltypes.ResultStream to iterate on.
-// This function should only be used for legacy code. New usage should directly use StreamExecute.
-func ExecuteWithStreamer(ctx context.Context, conn QueryService, target *querypb.Target, sql string, bindVariables map[string]*querypb.BindVariable, options *querypb.ExecuteOptions) sqltypes.ResultStream {
-	rs := &resultStreamer{
-		done: make(chan struct{}),
-		ch:   make(chan *sqltypes.Result),
-	}
-	go func() {
-		defer close(rs.done)
-		rs.err = conn.StreamExecute(ctx, target, sql, bindVariables, 0, 0, options, func(qr *sqltypes.Result) error {
-			select {
-			case <-ctx.Done():
-				return io.EOF
-			case rs.ch <- qr:
-			}
-			return nil
-		})
-		if rs.err == nil {
-			rs.err = io.EOF
-		}
-	}()
-	return rs
+type ReservedState struct {
+	ReservedID  int64
+	TabletAlias *topodatapb.TabletAlias
 }
 
-// ExecuteWithTransactionalStreamer does the same thing as ExecuteWithStreamer, but inside a transaction
-func ExecuteWithTransactionalStreamer(ctx context.Context, conn QueryService, target *querypb.Target, sql string, bindVariables map[string]*querypb.BindVariable, transactionID int64, options *querypb.ExecuteOptions) sqltypes.ResultStream {
-	rs := &resultStreamer{
-		done: make(chan struct{}),
-		ch:   make(chan *sqltypes.Result),
-	}
-	go func() {
-		defer close(rs.done)
-		rs.err = conn.StreamExecute(ctx, target, sql, bindVariables, transactionID, 0, options, func(qr *sqltypes.Result) error {
-			select {
-			case <-ctx.Done():
-				return io.EOF
-			case rs.ch <- qr:
-			}
-			return nil
-		})
-		if rs.err == nil {
-			rs.err = io.EOF
-		}
-	}()
-	return rs
```

Comment:
Why are we removing these?

Reply:
@frouioui This code looked to be entirely unused, so I cleaned it up while I was in here :smile:.

Reply:
Got it!